addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -290,63 +290,72 @@ def _l10n_it_is_simplified_document_type(self, document_type):
 
-            proxy_acks = []
-            for id_transaction, fattura in res.items():
-                if self.env['ir.attachment'].search([('name', '=', fattura['filename']), ('res_model', '=', 'account.move')], limit=1):
-                    # name should be unique, the invoice already exists
-                    _logger.info('E-invoice already exists: %s', fattura['filename'])
-                    proxy_acks.append(id_transaction)
-                    continue
+    def _save_incoming_attachment_fattura_pa(self, proxy_user, id_transaction, filename, content, key):
+        ''' Save an incoming file from the SdI as an attachment.
 
-                file = proxy_user._decrypt_data(fattura['file'], fattura['key'])
-
-                try:
-                    tree = etree.fromstring(file)
-                except Exception:
-                    # should not happen as the file has been checked by SdiCoop
-                    _logger.info('Received file badly formatted, skipping: \n %s', file)
-                    continue
-
-                attachment = self.env['ir.attachment'].create({
-                    'name': fattura['filename'],
-                    'raw': file,
-                    'type': 'binary',
-                })
+            :param proxy_user:     the user that saves the attachment.
+            :param id_transaction: id of the SdI transaction for communication with the IAP proxy.
+            :param filename:       name of the file to be saved.
+            :param content:        encrypted content of the file to be saved.
+            :param key:            key to decrypt the file.
+            :returns:              True if everything went well, or the file already exists.
+                                   False if the file cannot be parsed as an XML.
+        '''
 
-                if not self.env.context.get('test_skip_commit'):
-                    self.env.cr.commit() # In case something fails after, we still have the attachment
+        if self.env['ir.attachment'].search([('name', '=', filename), ('res_model', '=', 'account.move')], limit=1):
+            # name should be unique, the invoice already exists
+            _logger.info('E-invoice already exists: %s', filename)
+            return True
 
-                self.env['account.journal'].with_company(company)\
-                    .with_context(default_move_type='in_invoice')\
-                    ._create_document_from_attachment(attachment.ids)
+        decrypted_content = proxy_user._decrypt_data(content, key)
+        attachment = self.env['ir.attachment'].create({
+            'name': filename,
+            'raw': decrypted_content,
+            'type': 'binary'
+        })
 
-                proxy_acks.append(id_transaction)
-                if not self.env.context.get('test_skip_commit'):
-                    self.env.cr.commit()
+        # Failsafe commit: if something fails, the attachment is saved anyway.
+        self.env.cr.commit()
```

Comment:
But the fail-safe was removed in master in the sense that if something goes wrong, the file will never be imported and it will continue blocking on it.  (we created an invoice and unlinked it after)

Reply:
refs for discussion
https://github.com/odoo/odoo/blob/master/addons/account/models/account_journal.py#L746

Now the `get_edi_creation` method updates the invoice instead of creating it
https://github.com/odoo/odoo/blob/master/addons/account/models/account_move.py#LL2867C62-L2867C62